scalaz/src/main/scala/testz/z.scala

```diff
@@ -61,68 +61,78 @@ object z {
     F.tailrecM[(fold.S, unfold.S), B]({ case (fs, us) => go(fs, us) })((fold.start, unfold.start))
   }
 
-  object assertEqualNoShow {
-    def apply[E: Equal](fst: E, snd: E): List[TestError] =
-      if (fst === snd) Nil else Failure("not equal, when should be") :: Nil
+  implicit val equalTestResult: Equal[TestResult] = Equal.equal {
+    case (f, s) if f eq s => true
+    case (Success, Success) => true
+    case (f1: Failure, f2: Failure) => f1.failures == f2.failures
+    case _ => false // I don't care about comparing Throwable for equality.
```

Comment:
I would compare them using coalgebraic operations on `Throwable` (`getStackTrace`, `getClass` (because that's definitely part of `Throwable`'s identity), `getMessage`). That seems to be the only sane way to compare open classes, in terms of observable behavior through the parent class.

Reply:
The comment is wrong now. `failures` has throwables, so I *am* comparing them for equality.

I'm not sure what `equals` does; if it doesn't do what you're saying, I'll do it explicitly.